File: coco/pattern/sensec/app/demo/mock_jwt_validator.py
"""
Stand-in JWT validator for DEMO MODE ONLY.

Real deployments validate RS256-signed tokens against an identity provider
(see app.auth.jwt_validator.JWTValidator). For the demo, callers present a
plain bearer string from DEMO_TOKENS below — there is no signature, no
expiry, no issuer check. This exists purely so the UI can illustrate the
auth flow without standing up Azure AD.
"""

from __future__ import annotations

from app.auth.jwt_validator import TokenValidationError

DEMO_TOKENS: dict[str, dict[str, str]] = {
    "demo-token-payments-svc": {"sub": "demo-user-1", "app_id": "payments-svc"},
    "demo-token-reporting-app": {"sub": "demo-user-2", "app_id": "reporting-app"},
    "demo-token-ops-admin": {"sub": "demo-user-3", "app_id": "ops-admin"},
}

DEMO_SCOPES: dict[str, list[str]] = {
    "payments-svc": ["encrypt", "decrypt"],
    "reporting-app": ["decrypt"],
    "ops-admin": ["encrypt", "decrypt", "rotate", "grant", "manage_apps"],
}

# Seeded at startup: reporting-app may decrypt anything payments-svc encrypted.
DEMO_GRANTS: list[tuple[str, str]] = [
    ("reporting-app", "payments-svc"),
]


class MockJWTValidator:
    def validate(self, token: str) -> dict[str, str]:
        claims = DEMO_TOKENS.get(token)
        if claims is None:
            raise TokenValidationError("Unknown demo token")
        return claims

File: coco/pattern/springsec/hsm-core-service/src/main/java/com/hsm/core/auth/MockJwtValidator.java
package com.hsm.core.auth;

import java.util.HashMap;
import java.util.List;
import java.util.Map;

/**
 * Fake JWT validation for demo mode -- a fixed lookup table, no signature/expiry/
 * issuer verification. Ported from app/demo/mock_jwt_validator.py. Also carries
 * the demo app scope/grant seed data consumed at startup by DemoSeedInitializer.
 */
public class MockJwtValidator implements JwtValidator {

    public static final Map<String, Map<String, String>> DEMO_TOKENS = Map.of(
            "demo-token-payments-svc", Map.of("sub", "demo-user-1", "app_id", "payments-svc"),
            "demo-token-reporting-app", Map.of("sub", "demo-user-2", "app_id", "reporting-app"),
            "demo-token-ops-admin", Map.of("sub", "demo-user-3", "app_id", "ops-admin")
    );

    public static final Map<String, List<String>> DEMO_SCOPES = Map.of(
            "payments-svc", List.of("encrypt", "decrypt", "dek_issue", "dek_unwrap"),
            "reporting-app", List.of("decrypt"),
            "ops-admin", List.of("encrypt", "decrypt", "rotate", "grant", "manage_apps", "provision_app_keys")
    );

    /** (granteeAppId, ownerAppId) pairs. reporting-app may decrypt anything payments-svc encrypted. */
    public static final List<Map.Entry<String, String>> DEMO_GRANTS = List.of(
            Map.entry("reporting-app", "payments-svc")
    );

    @Override
    public Map<String, Object> validate(String token) throws TokenValidationException {
        Map<String, String> claims = DEMO_TOKENS.get(token);
        if (claims == null) {
            throw new TokenValidationException("Unknown demo token");
        }
        Map<String, Object> result = new HashMap<>();
        result.putAll(claims);
        return result;
    }
}
